react-basic-practice/src/components/common/Input/index.jsx

```diff
@@ -1,37 +1,52 @@
-import React from 'react';
+import React, { useRef } from 'react';
 import './index.css';
 
 const Input = (props) => {
   const {
     label,
     htmlFor,
     inputType,
-    inputId,
+    name,
     cssClasses,
     value,
-    refer,
-    onChange,
     placeholder,
     pattern,
     defaultValue,
+    handleInput,
+    messageErr,
   } = props;
 
+  const inputRef = useRef();
+
+  const handleChange = (e) => {
+    const inputValue = { [e.target.name]: inputRef.current.value };
+
+    handleInput(inputValue);
```

Comment:
`handleInput` is too vague. Should specify what event you are handling: `handleInputChange`.